### ISSUE TITLE
examples/image_classification-tf.ipynb What version of tensor flow should I use

### ISSUE THREAD

**Issue by ERIK54600** [huggingface/notebooks#516]
While running on my local, I get the the following error, not sure what version of tensor flow should I be using,
<img width="652" alt="Screenshot 2024-09-18 at 8 27 16 AM" src="https://github.com/user-attachments/assets/94e1f26c-06ef-4818-80b2-6f901901cdb4">
I am using the following libraries

# Name                    Version                   Build  Channel
_tflow_select             2.2.0                     eigen  
abseil-cpp                20210324.2           hbdafb3b_0    conda-forge
absl-py                   2.1.0            py38hca03da5_0  
aiohappyeyeballs          2.4.0            py38hca03da5_0  
aiohttp                   3.10.5           py38h80987f9_0  
aiosignal                 1.2.0              pyhd3eb1b0_0  
appnope                   0.1.2           py38hca03da5_1001  
arrow-cpp                 8.0.0           py38hacdd530_1_cpu    conda-forge
asttokens                 2.0.5              pyhd3eb1b0_0  
astunparse                1.6.3                      py_0  
async-timeout             4.0.3            py38hca03da5_0  
attrs                     23.1.0           py38hca03da5_0  
aws-c-cal                 0.5.11               h4530763_0    conda-forge
aws-c-common              0.6.2                h3422bc3_0    conda-forge
aws-c-event-stream        0.2.7               h9972306_13    conda-forge
aws-c-io                  0.10.5               hea86ef8_0    conda-forge
aws-checksums             0.1.11               h487e1a8_7    conda-forge
aws-sdk-cpp               1.8.186              h392f50b_4    conda-forge
backcall                  0.2.0              pyhd3eb1b0_0  
blas                      1.0                    openblas  
blinker                   1.6.2            py38hca03da5_0  
boost-cpp                 1.81.0               hf96b251_0    conda-forge
bottleneck                1.3.2            py38h691f20f_4    conda-forge
brotli                    1.0.9                h1a8c8d9_9    conda-forge
brotli-bin                1.0.9                h1a8c8d9_9    conda-forge
brotli-python             1.0.9            py38h313beb8_8  
bzip2                     1.0.8                h99b78c6_7    conda-forge
c-ares                    1.19.1               h80987f9_0  
ca-certificates           2024.8.30            hf0a4a13_0    conda-forge
cachetools                5.3.3            py38hca03da5_0  
certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge
cffi                      1.16.0           py38h80987f9_1  
charset-normalizer        3.3.2              pyhd3eb1b0_0  
click                     8.1.7            py38hca03da5_0  
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
comm                      0.2.1            py38hca03da5_0  
contourpy                 1.1.1            py38h9afee92_1    conda-forge
cryptography              41.0.3           py38h3c57c4d_0  
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
dataclasses               0.8                pyhc8e2a94_3    conda-forge
datasets                  2.16.1             pyhd8ed1ab_0    conda-forge
debugpy                   1.6.7            py38h313beb8_0  
decorator                 5.1.1              pyhd3eb1b0_0  
dill                      0.3.6              pyhd8ed1ab_1    conda-forge
executing                 0.8.3              pyhd3eb1b0_0  
filelock                  3.16.1             pyhd8ed1ab_0    conda-forge
flatbuffers               2.0.8                hb7217d7_1    conda-forge
fonttools                 4.53.1           py38h3237794_0    conda-forge
freetype                  2.12.1               hadb7bae_2    conda-forge
frozenlist                1.4.0            py38h80987f9_0  
fsspec                    2023.10.0          pyhca7485f_0    conda-forge
gast                      0.4.0              pyhd3eb1b0_0  
gflags                    2.2.2             hc88da5d_1004    conda-forge
giflib                    5.2.1                h80987f9_3  
glog                      0.6.0                h6da1cb0_0    conda-forge
google-auth               2.29.0           py38hca03da5_0  
google-auth-oauthlib      0.4.4              pyhd3eb1b0_0  
google-pasta              0.2.0              pyhd3eb1b0_0  
grpc-cpp                  1.43.2               ha558326_3    conda-forge
grpcio                    1.43.0           py38h69ee544_0    conda-forge
h5py                      3.11.0           py38haafd478_0  
hdf5                      1.12.1               h160e8cb_2  
huggingface_hub           0.24.7             pyhd8ed1ab_0    conda-forge
icu                       70.1                 h6b3803e_0    conda-forge
idna                      3.7              py38hca03da5_0  
importlib-metadata        7.0.1            py38hca03da5_0  
importlib-resources       6.4.5              pyhd8ed1ab_0    conda-forge
importlib_metadata        7.0.1                hd3eb1b0_0  
importlib_resources       6.4.5              pyhd8ed1ab_0    conda-forge
ipykernel                 6.28.0           py38hca03da5_0  
ipython                   8.12.2           py38hca03da5_0  
jedi                      0.19.1           py38hca03da5_0  
joblib                    1.4.2              pyhd8ed1ab_0    conda-forge
jpeg                      9e                   h80987f9_3  
jupyter_client            8.6.0            py38hca03da5_0  
jupyter_core              5.7.2            py38hca03da5_0  
keras                     2.8.0              pyhd8ed1ab_0    conda-forge
keras-preprocessing       1.1.2              pyhd3eb1b0_0  
kiwisolver                1.4.5            py38h9afee92_1    conda-forge
krb5                      1.20.1               h8380606_1  
lcms2                     2.15                 h481adae_0    conda-forge
lerc                      4.0.0                h9a09cb3_0    conda-forge
libbrotlicommon           1.0.9                h1a8c8d9_9    conda-forge
libbrotlidec              1.0.9                h1a8c8d9_9    conda-forge
libbrotlienc              1.0.9                h1a8c8d9_9    conda-forge
libcrc32c                 1.1.2                hbdafb3b_0    conda-forge
libcurl                   8.2.1                h0f1d93c_0  
libcxx                    18.1.8               h3ed4263_7    conda-forge
libdeflate                1.17                 h1a8c8d9_0    conda-forge
libedit                   3.1.20230828         h80987f9_0  
libev                     4.33                 h1a28f6b_1  
libevent                  2.1.10               hbae9a57_4    conda-forge
libffi                    3.4.4                hca03da5_1  
libgfortran               5.0.0           11_3_0_hca03da5_28  
libgfortran5              11.3.0              h009349e_28  
libgoogle-cloud           1.36.0               h2b0db34_0    conda-forge
libnghttp2                1.52.0               h10c0552_1  
libopenblas               0.3.21               h269037a_0  
libpng                    1.6.39               h80987f9_0  
libprotobuf               3.19.6               hb5ab8b9_0    conda-forge
libsodium                 1.0.18               h1a28f6b_0  
libsqlite                 3.46.0               hfb93653_0    conda-forge
libssh2                   1.10.0               h449679c_2  
libthrift                 0.16.0               h1a74c4f_2    conda-forge
libtiff                   4.5.0                h5dffbdd_2    conda-forge
libutf8proc               2.8.0                h1a8c8d9_0    conda-forge
libwebp-base              1.4.0                h93a5062_0    conda-forge
libxcb                    1.13              h9b22ae9_1004    conda-forge
libzlib                   1.2.13               hfb2fe0b_6    conda-forge
llvm-openmp               18.1.8               hde57baf_1    conda-forge
lz4-c                     1.9.3                hbdafb3b_1    conda-forge
markdown                  3.4.1            py38hca03da5_0  
markupsafe                2.1.3            py38h80987f9_0  
matplotlib                3.7.2            py38h150bfb4_0    conda-forge
matplotlib-base           3.7.2            py38h0405b42_0    conda-forge
matplotlib-inline         0.1.6            py38hca03da5_0  
multidict                 6.0.4            py38h80987f9_0  
multiprocess              0.70.12.2        py38hea4295b_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
ncurses                   6.4                  h313beb8_0  
nest-asyncio              1.6.0            py38hca03da5_0  
numexpr                   2.7.3            py38h3777fb4_0    conda-forge
numpy                     1.24.3           py38h1398885_0  
numpy-base                1.24.3           py38h90707a3_0  
oauthlib                  3.2.2            py38hca03da5_0  
openjpeg                  2.5.0                hbc2ba62_2    conda-forge
openssl                   1.1.1w               h53f4e23_0    conda-forge
opt_einsum                3.3.0              pyhd3eb1b0_1  
orc                       1.7.3                hcb6706d_0    conda-forge
packaging                 24.1             py38hca03da5_0  
pandas                    2.0.3            py38h46d7db6_0  
parquet-cpp               1.5.1                         2    conda-forge
parso                     0.8.3              pyhd3eb1b0_0  
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    9.4.0            py38h1bb68ce_1    conda-forge
pip                       24.2             py38hca03da5_0  
platformdirs              3.10.0           py38hca03da5_0  
pooch                     1.8.2              pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.43           py38hca03da5_0  
protobuf                  3.19.6           py38h2b1e499_0    conda-forge
psutil                    5.9.0            py38h1a28f6b_0  
pthread-stubs             0.4               h27ca646_1001    conda-forge
ptyprocess                0.7.0              pyhd3eb1b0_2  
pure_eval                 0.2.2              pyhd3eb1b0_0  
pyarrow                   8.0.0           py38h535b32c_1_cpu    conda-forge
pyarrow-hotfix            0.6                pyhd8ed1ab_0    conda-forge
pyasn1                    0.4.8              pyhd3eb1b0_0  
pyasn1-modules            0.2.8                      py_0  
pycparser                 2.21               pyhd3eb1b0_0  
pygments                  2.15.1           py38hca03da5_1  
pyjwt                     2.8.0            py38hca03da5_0  
pyopenssl                 23.2.0           py38hca03da5_0  
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1            py38hca03da5_0  
python                    3.8.15          hf452327_0_cpython    conda-forge
python-dateutil           2.9.0post0       py38hca03da5_2  
python-flatbuffers        24.3.25          py38hca03da5_0  
python-tzdata             2024.1             pyhd8ed1ab_0    conda-forge
python-xxhash             2.0.2            py38hea4295b_0    conda-forge
python_abi                3.8                      2_cp38    conda-forge
pytz                      2024.2             pyhd8ed1ab_0    conda-forge
pyyaml                    6.0              py38hea4295b_0    conda-forge
pyzmq                     25.1.2           py38h313beb8_0  
re2                       2022.02.01           hbdafb3b_0    conda-forge
readline                  8.2                  h1a28f6b_0  
regex                     2021.10.23       py38hea4295b_0    conda-forge
requests                  2.32.3           py38hca03da5_0  
requests-oauthlib         2.0.0            py38hca03da5_0  
responses                 0.18.0             pyhd8ed1ab_0    conda-forge
rsa                       4.7.2              pyhd3eb1b0_1  
safetensors               0.4.4            py38h186058e_0    conda-forge
scikit-learn              1.3.2            py38he1bc1c9_2    conda-forge
scipy                     1.10.1           py38h9d039d2_1  
setuptools                72.1.0           py38hca03da5_0  
six                       1.16.0             pyhd3eb1b0_1  
snappy                    1.1.10               hd04f947_1    conda-forge
sqlite                    3.45.3               h80987f9_0  
stack_data                0.2.0              pyhd3eb1b0_0  
tensorboard               2.8.0              pyhd8ed1ab_1    conda-forge
tensorboard-data-server   0.6.1            py38ha6e5c4f_0  
tensorboard-plugin-wit    1.8.1            py38hca03da5_0  
tensorflow                2.8.0           cpu_py38h4487131_0    conda-forge
tensorflow-base           2.8.0           cpu_py38h0aa0dbe_0    conda-forge
tensorflow-estimator      2.8.0           cpu_py38h888fe13_0    conda-forge
termcolor                 2.1.0            py38hca03da5_0  
threadpoolctl             3.5.0              pyhc1e730c_0    conda-forge
tk                        8.6.14               h6ba3021_0  
tokenizers                0.19.1           py38h92154f2_0  
tornado                   6.4.1            py38h80987f9_0  
tqdm                      4.66.5             pyhd8ed1ab_0    conda-forge
traitlets                 5.14.3           py38hca03da5_0  
transformers              4.44.2             pyhd8ed1ab_0    conda-forge
typing-extensions         4.11.0           py38hca03da5_0  
typing_extensions         4.11.0           py38hca03da5_0  
unicodedata2              15.1.0           py38hb192615_0    conda-forge
urllib3                   2.2.2            py38hca03da5_0  
utf8proc                  2.6.1                h80987f9_1  
wcwidth                   0.2.5              pyhd3eb1b0_0  
werkzeug                  3.0.3            py38hca03da5_0  
wheel                     0.35.1             pyhd3eb1b0_0  
wrapt                     1.14.1           py38h1a28f6b_0  
xorg-libxau               1.0.11               hb547adb_0    conda-forge
xorg-libxdmcp             1.1.3                h27ca646_0    conda-forge
xxhash                    0.8.0                h27ca646_3    conda-forge
xz                        5.4.6                h80987f9_1  
yaml                      0.2.5                h3422bc3_2    conda-forge
yarl                      1.11.0           py38h80987f9_0  
zeromq                    4.3.5                h313beb8_0  
zipp                      3.17.0           py38hca03da5_0  
zlib                      1.2.13               hfb2fe0b_6    conda-forge
zstd                      1.5.6                hb46c0d2_0    conda-forge

**Comment by techyorzl** [comment]
Its not a problem with Tensorflow version as for these type of tasks, Tensorflow 2.x (specifically >2.4 is preferred. I would suugest you do some updates with what code you have written. I can help you with the snippet if this problem still persists!
